### ISSUE TITLE
force:data:soql:query returns [object Object] when querying grandparent fields

### ISSUE THREAD

**Issue by sjurgis** [forcedotcom/cli#1164]
### Summary

Queries that reference grand parent fields (say Lead owner profile name as in example below) does not render requested field correctly.

Looking at --json version the field is returned, but not rendered correctly.

### Steps To Reproduce:
```
sfdx force:data:soql:query -q 'select owner.Profile.Name from lead limit 1'
```
### Expected result

```
OWNER.PROFILE
───────────────
System Administrator
```

### Actual result

```
OWNER.PROFILE
───────────────
[object Object]
```

### System Information

sfdx-cli/7.116.2 darwin-x64 node-v14.15.1

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-9881177](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000005voF1YAI/view)

**Comment by MuriloKakazu** [comment]
Related to this closed issue from last year: https://github.com/forcedotcom/cli/issues/264

The original issue was closed but the bug was never fixed :(

**Comment by cristiand391** [comment]
This was fixed in `plugin-data@0.6.2` which has been included in sfdx since v7.121.3: https://github.com/salesforcecli/sfdx-cli/commit/4e109e3ef580e8443a2a85e7bb8da90df10ecdd7
